datenguide_python/output_transformer.py

```diff
@@ -1,82 +1,95 @@
+import pandas as pd
+
+import numpy as np
+
+from pandas.io.json import json_normalize
+
+
 class QueryOutputTransformer:
     """ IN PROGRESS only region query to do: DOKU """
 
     def __init__(self):
         pass
 
     @staticmethod
-    def flatten_query(dic):
-        from pandas.io.json import json_normalize
-
+    def flatten_query_result(dic):
         flat = json_normalize(dic)
         return flat
 
     @staticmethod
     def get_col_list(df):
         c = list(df.columns)
-        c.remove("data.region.id")
-        c.remove("data.region.name")
+        try:
+            c.remove("data.region.id")
+        except ValueError:
+            pass
+        try:
+            c.remove("data.region.name")
+        except ValueError:
+            pass
         return c
 
     @staticmethod
-    def build_and_merge_data(dic, lis):
-        import pandas as pd
-        from pandas.io.json import json_normalize
+    def build_and_merge_data(dic, lis, year_default=True):
+        data = json_normalize(dic[lis[0]][0])
+        if year_default is True:
+            data.columns = [lis[0], "year"]
+            for d in lis[1:]:
+                if json_normalize(dic[d][0]).empty is True:
+                    temp = pd.DataFrame([np.nan, np.nan]).T
+                    temp.columns = [d, "year"]
+                elif json_normalize(dic[d][0]).empty is False:
+                    temp = json_normalize(dic[d][0])
+                    try:
+                        temp.columns = [d, "year"]
+                        print("no type variable")
+                    except ValueError:
+                        newcolname = temp.values[0][0] + "_" + d
+                        temp = temp.iloc[:, 1:]
+                        temp.columns = [newcolname, "year"]
+                        print("type variable")
+                data = data.merge(temp, on="year", how="outer")
 
-        data = pd.DataFrame()
-        for d in lis:
+            newcol = [x.replace("data.region.", "") for x in data.columns]
+            data.columns = newcol
             try:
-                if data.empty is True:
-                    data = json_normalize(dic[d][0])
-                    data.columns = [d, "year"]
-                else:
-                    temp = json_normalize(dic[d][0])
-                    temp.columns = [d, "year"]
-                    data = data.merge(temp, on="year", how="outer")
-                print("no type variable")
+                data["id"] = dic["data.region.id"][0]
```

Comment:
Appears inconsistent as the json dictionary does not have lists for `id` and `name`. Is this related this function being called with a DataFrame instead of a dictionary?